spec/features/user_interacts_with_queue_spec.rb

```diff
@@ -0,0 +1,58 @@
+require "spec_helper"
+
+feature "User interacts with the queue" do 
+  scenario "user adds and reorders videos in the queue" do 
+    comedies = Fabricate(:category)
+    monk = Fabricate(:video, title: "Monk", category: comedies)
+    south_park = Fabricate(:video, title: "South Park", category: comedies)
+    futurama = Fabricate(:video, title: "Futurama", category: comedies)
+    
+    sign_in 
+    add_video_to_queue(monk)
+    expect_video_to_be_in_queue(monk)
+    
+    visit video_path(monk)
+    expect_link_not_to_be_seen("+ My Queue")
+
+    add_video_to_queue(south_park)
+    add_video_to_queue(futurama)
+    
+    set_video_position(monk, 3)
+    set_video_position(south_park, 1)
+    set_video_position(futurama, 2)
+    
+    update_queue
+    
+    expect_video_position(monk, 3)
+    expect_video_position(south_park, 1)
+    expect_video_position(futurama, 2)
+  end
+  
+  def update_queue
+    click_button "Update Instant Queue"
+  end
+  
+  def expect_link_not_to_be_seen(link_text)
+    page.should_not have_content link_text 
```

Comment:
`should` syntax is not recommended and it will be deprecated in the future and possibly moved to a separate gem. You should be using `expect` syntax. Also, the opposite of `to have_content` is actually `to have_no_content`.

http://www.rubydoc.info/github/jnicklas/capybara/master/Capybara/Node/Matchers#has_no_text%3F-instance_method